src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,168 @@
+        LOAD,
+        PLAY
+    }
+
+    @EventHandler
+    val listener = Listener<TickEvent.InGame>({
+        val player = it.player
+        val world = it.world
+
+        when (mode) {
+            NotebotMode.DISCOVER -> {
+                for (x in -4..4) for (y in -4..4) for (z in -4..4) {
+                    val pos = player.pos?.plus(Vec3d(x.toDouble(), y.toDouble(), z.toDouble())) ?: break
+                    if (world.getBlockState(BlockPos(pos))?.block == Blocks.NOTE_BLOCK) {
+                        val noteBlock = world.getBlockState(BlockPos(pos))
+                        snackbarMessage(
+                            player, ""+noteBlock.get(INSTRUMENT) + "[" + noteBlock.get(NOTE) + "]"
+                        )
+                        map.add(noteBlock.get(INSTRUMENT), noteBlock.get(NOTE), BlockPos(pos))
+                    }
+                }
+            }
+            NotebotMode.LOAD -> {
+                noteSequence = MidiParser.parse("songs" + File.separator + songName, 5)
+                if (noteSequence.isEmpty())
+                    snackbarMessage(player, "Song not found: " + "songs" + File.separator + songName)
+                else
+                    snackbarMessage(player, "Song Found: " + "songs" + File.separator + songName)
+                lastNote = System.currentTimeMillis()
+                elapsed = 0
+            }
+            NotebotMode.PLAY -> {
+                if (!player.isCreative) {
+                    while (noteSequence.isNotEmpty() && noteSequence.firstKey() <= elapsed) {
+                        playNotes(noteSequence.pollFirstEntry().value, player, world)
+                    }
+                    elapsed += System.currentTimeMillis() - lastNote
+                    lastNote = System.currentTimeMillis()
+                } else snackbarMessage(player, "You are in creative mode and cannot play music.")
+            }
+        }
+    })
+
+    private fun playNotes(notes: ArrayList<Note>, player: ClientPlayerEntity, world: ClientWorld) {
+        var blockPosArr: ArrayList<BlockPos?> = ArrayList()
+        notes.forEach { n ->
+            channelsArray = arrayOf(ChannelZero, ChannelOne, ChannelTwo, ChannelThree, ChannelFour)
+            val enum = channelsArray[Math.abs(n.track % (channelsArray.size - 1))]
+            snackbarMessage(player, "" + enum)
+            blockPosArr.add(map.get(enum)[if (enum == Instrument.HAT && hatAlwaysAsFSharp) 0 else n.notebotNote])
+        }
+        playBlock(blockPosArr, player, world)
+    }
+
+    fun snackbarMessage(player: ClientPlayerEntity, t: String) {
+        player.sendMessage(text { +t }, true)
+    }
+
+    private fun playBlock(blockPosTracks: ArrayList<BlockPos?>, player: ClientPlayerEntity, world: ClientWorld) {
+        if (blockPosTracks.isNotEmpty()) {
+            blockPosTracks.forEach { blockPos ->
+                if (blockPos != null) {
+                    faceVectorPacketInstant(player, blockPos.asVec3d, mc)
```

Comment:
Modify the packet instead, sending extra packets will flags anticheat

Reply:
This means we can't play notes fast enough.

Reply:
Why'd you use notebot when there's an anti cheat tho? I mean the only purpose is pretty much for bragging and then you clearly aren't trying to hide the use of utility mods in that case anyways.

Reply:
This is an unhelpful comment. The purpose of Notebot would be to use it on a server you own, or that you are allowed to use it on.

Reply:
Well then you'd be excluded from anti cheat and you wouldn't have to worry about spamming packets.

Reply:
This conversation is pointless. KAMI isn't just for use on servers with AC